xc/device.ml

```diff
@@ -1270,18 +1317,29 @@ module Varstored = SystemdDaemonMgmt (struct
   let use_pidfile = true
 
   let pid_path domid = sprintf "/local/domain/%d/varstored-pid" domid
+
+  let expected_cmdline_items ~domid =
+      [!Xc_resources.varstored; pid_path domid]
 end)
 
 module PV_Vnc = struct
+
+  let vnc_console_path domid = sprintf "/local/domain/%d/console" domid
+
   module D = DaemonMgmt (struct
     let name = "vncterm"
 
     let use_pidfile = false
 
     let pid_path domid = sprintf "/local/domain/%d/vncterm-pid" domid
+
+    let expected_cmdline_items ~domid =
+      [
+        !Xc_resources.vncterm (* vncterm binary path *)
+      ; vnc_console_path domid (* xenstore console path *)
+      ]
   end)
 
-  let vnc_console_path domid = sprintf "/local/domain/%d/console" domid
 
   let pid ~xs domid = D.pid ~xs domid
 
```

Comment:
In the code change(https://github.com/xapi-project/xen-api/commit/9ea53a0d033d1f76b1d2f35e59d7b61669db7d46#diff-bbd679d3846d8116c11662869402a1202e656b6611a2727e2ece98bb637080ac) from xen-api: the 2 functions  `is_cmdline_valid` and `is_vncterm_running` are removed, and in `get_vnc_port` and `get_tc_port`, `D.is_running` is used to replace `is_vncterm_running`, so that the ability to verify commandline in the base module got used here.

May I know why we do not change it here, is there any considerations? Thank you~

Reply:
Could be an oversight. I had to port this manually because the code is organised differently. Will take a look

Reply:
OK, here is the piece of that code change:
```
@@ -825,36 +886,15 @@ module PV_Vnc = struct
 
   let pid ~xs domid = D.pid ~xs domid
 
-  (* Look up the commandline args for the vncterm pid; *)
-  (* Check that they include the vncterm binary path and the xenstore console
-     path for the supplied domid. *)
-  let is_cmdline_valid domid pid =
-    try
-      let cmdline =
-        Printf.sprintf "/proc/%d/cmdline" pid
-        |> Unixext.string_of_file
-        |> Astring.String.cuts ~sep:"\000"
-      in
-      List.mem !Xc_resources.vncterm cmdline
-      && List.mem (vnc_console_path domid) cmdline
-    with _ -> false
-
-  let is_vncterm_running ~xs domid =
-    match pid ~xs domid with
-    | None ->
-        false
-    | Some p ->
-        D.is_running ~xs domid && is_cmdline_valid domid p
-
   let get_vnc_port ~xs domid =
-    if not (is_vncterm_running ~xs domid) then
+    if not (D.is_running ~xs domid) then
       None
     else
       try Some (Socket.Port (int_of_string (xs.Xs.read (vnc_port_path domid))))
       with _ -> None
 
   let get_tc_port ~xs domid =
-    if not (is_vncterm_running ~xs domid) then
+    if not (D.is_running ~xs domid) then
       None

```

Reply:
I've added this.